Sources/Lumina/UI/Extensions/InterfaceHandlerExtension.swift

```diff
@@ -0,0 +1,135 @@
+    }
+  }
+
+  func updateUI(orientation: UIInterfaceOrientation) {
+    LuminaLogger.notice(message: "updating UI for orientation: \(orientation.rawValue)")
+    guard let connection = self.previewLayer.connection, connection.isVideoOrientationSupported else {
+      return
+    }
+    self.previewLayer.frame = self.view.bounds
+    connection.videoOrientation = necessaryVideoOrientation(for: orientation)
+    self.camera?.updateOutputVideoOrientation(connection.videoOrientation)
+  }
+
+  func updateButtonFrames() {
+    let frame = self.view.safeAreaLayoutGuide.layoutFrame
+    self.switchButton.center = CGPoint(x: frame.maxX - 30, y: frame.minY + 25)
+    self.cancelButton.center = CGPoint(x: frame.minX + 55, y: frame.maxY - 45)
+    self.torchButton.center = CGPoint(x: frame.minX + 25, y: frame.minY + 25)
+    self.shutterButton.center = CGPoint(x: frame.midX, y: frame.maxY - 45)
+
+    let textWidth = frame.maxX - 110
+    self.textPromptView.frame.size = CGSize(width: textWidth - 10, height: 80)
+    self.textPromptView.layoutSubviews()
+    self.textPromptView.center = CGPoint(x: frame.midX, y: frame.minY + 45)
+  }
+
+  // swiftlint:disable cyclomatic_complexity
+  func handleCameraSetupResult(_ result: CameraSetupResult) {
+    LuminaLogger.notice(message: "camera set up result: \(result.rawValue)")
+    DispatchQueue.main.async {
+      switch result {
+        case .videoSuccess:
+          if let camera = self.camera {
+            self.enableUI(valid: true)
+            self.updateUI(orientation: UIApplication.shared.statusBarOrientation)
+            camera.start()
+          }
+        case .audioSuccess:
+          break
+        case .requiresUpdate:
+          self.camera?.updateVideo({ result in
+            self.handleCameraSetupResult(result)
+          })
+        case .videoPermissionDenied:
+          self.textPrompt = "Camera permissions for Lumina have been previously denied - please access your privacy settings to change this."
+        case .videoPermissionRestricted:
+          self.textPrompt = "Camera permissions for Lumina have been restricted - please access your privacy settings to change this."
+        case .videoRequiresAuthorization:
+          self.camera?.requestVideoPermissions()
+        case .audioPermissionRestricted:
+          self.textPrompt = "Audio permissions for Lumina have been restricted - please access your privacy settings to change this."
+          DispatchQueue.main.asyncAfter(deadline: .now() + 2) {
+            self.textPrompt = ""
+          }
+        case .audioRequiresAuthorization:
+          self.camera?.requestAudioPermissions()
+        case .audioPermissionDenied:
+          self.textPrompt = "Audio permissions for Lumina have been previously denied - please access your privacy settings to change this."
+          DispatchQueue.main.asyncAfter(deadline: .now() + 2) {
+            self.textPrompt = ""
+          }
+        case .invalidVideoDataOutput,
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/dokun1/Lumina/blob/f3c4aa8d5c66dd9d2660cdba8e658b8902c9752f/Sources/Lumina/UI/Extensions/InterfaceHandlerExtension.swift#L107"></span>Case statements should vertically align with their enclosing switch statement.<br />
<code>switch_case_alignment</code> <code>InterfaceHandlerExtension.swift:107</code></td>
    </tr>
  </tbody>
</table>